T3/Gui/Graph/Interaction/SymbolBrowser.cs

```diff
@@ -76,62 +76,68 @@ public void Draw()
 
             ImGui.PushID(UiId);
             {
-                _posInWindow = GraphCanvas.Current.ChildPosFromCanvas(PosOnCanvas);
+                var posInWindow = GraphCanvas.Current.ChildPosFromCanvas(PosOnCanvas);
                 _posInScreen = GraphCanvas.Current.TransformPosition(PosOnCanvas);
                 _drawList = ImGui.GetWindowDrawList();
 
                 ImGui.SetNextWindowFocus();
-                DrawMatchesList();
-                DrawSearchInput();
+
+                Vector2 browserPositionInWindow;
+                ShiftPositionToFitOnCanvas(out browserPositionInWindow, posInWindow + _browserPositionOffset, _resultListSize);
```

Comment:
Nice solution. But why an out param instead of a Vector2 return value?

Reply:
just because the description of what it is doing - if it's shifting something, a return value doesn't necessarily suggest that as intuitively as passing in a reference to something that will be shifted

Reply:
but even by my own description that's not really any clearer and i should just be passing a `ref` if i want to go that route

that function is under construction per your feedback and i'll keep this in mind